### DISCUSSION TITLE
direct pars check_diff() failing

### DISCUSSION THREAD

**Discussion by wkitlasten** [pypest/pyemu#425] [General]
I am using pst_from.add_parameters() to add direct parameters for each layer (k) for zones defined by gclass[k]. pf makes it through layers 1, 10, 11, 12, 13, 2, 3, 4, 5... and then crashes on layer 6. Sounds like an issue with the arrays, right? I tried to find a difference, but just can't see it. I have attached layers 5 and 6 input arrays (arr_f) and gclass arrays in case I am just being dense and there is something obviously wrong there. Any insights appreciated!

My call:
```
pf.add_parameters(filenames=arr_f, par_type="zone", zone_array=gclass[k],
                              par_name_base=pargp, pargp=pargp,
                              upper_bound=ubound, lower_bound=lbound,
                              ult_ubound=ult_ubound, ult_lbound=ult_lbound,
                              par_style='direct', transform='none', )
```

The error:
 ```
 File "D:\modelling\pyemu\pyemu\utils\pst_from.py", line 2217, in add_parameters
    df = write_array_tpl(
  File "D:\modelling\pyemu\pyemu\utils\pst_from.py", line 3745, in write_array_tpl
    _check_diff(zone_org_arr, input_filename, zval)
  File "D:\modelling\pyemu\pyemu\utils\pst_from.py", line 3859, in _check_diff
    raise Exception(message)
Exception: _check_diff() error: direct par for file 'C:\wairau\wairau\waq_14_s1_200_ies\org\waq_14_s1_200.dimtortuosity_layer6.arr'exceeds tolerance for percent difference: 16.669935281427733 > 1.0 in zone 194
```

**Comment by jtwhite79** [comment]
what check diff is doing is trying to make sure that values of the array in each zone are uniform for direct pars...so if you load up that zone array and pull out the layer 6 array values in each zone, they are uniform?

**Reply by wkitlasten** [reply]
1 stray cell with a different value... how'd that happen!! Thanks

**Reply by jtwhite79** [reply]
#modeling
